Document standard metadata entries. (#61941)
elastic/elasticsearch: docs/reference/mapping/params/meta.asciidoc

NOTE: Field metadata is updatable by submitting a mapping update. The metadata
of the update will override the metadata of the existing field.

Elastic products standardize on the following metadata, which you may use too
in order to get a better out-of-the-box experience with your data.

unit::

  The unit of the field. Recognizes `"byte"` as well as
  <<time-units,time units>>. By default, fields are assumed to not have a unit.
  This information should only be specified on numeric fields.

metric_type::

  The type of metric. Recognizes `"gauge"` and `"counter"`. By default, fields
  are assumed to not be a metric. This information should only be specified on

>   numeric fields.

Reviewer: I'd like to explain what gauges and counters are in here. Something like:

The type of metric for time series data. Options are none, `"gauge"` and `"counter"`.

A gauge is a number representing a point in time that can go up and down.

A counter is a number representing a cumulative amount and is always increasing. Counters can reset to 0 occasionally.

ratio::

>   Whether the field is a ratio. Recognizes `true` and `false` (default). The

Reviewer: ```suggestion
  Whether the field is a ratio or percent. Recognizes `true` and `false` (default). The
```

Author: My understanding was that we always use ratios in the data we produce, never percentages (ie. ratio * 100), is this assumption wrong? If we have both, do we need to be able to make the distinction via this metadata so that Kibana would know?

Reviewer 2: @roncohen @ruflin and I had a small discussion about this yesterday. We see how this can be confusing for users, so we agreed this could be the best approach:

Having only `percent` stored as a unit (no need for a ratio flag in meta), where the value is expected to be in the [0-1] notation. This is how Metricbeat stores percentages today.

WDYT?

NOTE: Field metadata is updatable by submitting a mapping update. The metadata
of the update will override the metadata of the existing field.

> Elastic products standardize on the following metadata, which you may use too
> in order to get a better out-of-the-box experience with your data.

Reviewer 3: ```suggestion
Elastic products use the following standard metadata entries for fields. You  can follow these same metadata conventions to get a better out-of-the-box experience with your data. 
```

unit::

>   The unit of the field. Recognizes `"byte"` as well as
>   <<time-units,time units>>. By default, fields are assumed to not have a unit.
>   This information may only be specified on numeric fields.

Reviewer 3: ```suggestion
  The unit associated with a numeric field: `"byte"` or a
  <<time-units,time unit>>. By default, a field does not have a unit.
  Only valid for numeric fields.
```

metric_type::

>   The type of metric. Recognizes `"gauge"` and `"counter"`. A gauge is a
>   single-valued measurement that can go up or down over time, such as a
>   temperature. A counter is a single-valued cumulative counter that only goes
>   up, such as the number of requests processed by a web server. By default,
>   fields are assumed to not be a metric. This information may only be
>   specified on numeric fields.

Reviewer 3: ```suggestion
  The  metric type of a numeric field: `"gauge"` or `"counter"`. A gauge is a
  single-value measurement that can go up or down over time, such as a
  temperature. A counter is a single-value cumulative counter that only goes
  up, such as the number of requests processed by a web server. By default,
 no metric type is associated with a field. Only valid for numeric fields.
```

ratio::

>   Whether the field is a ratio. Recognizes `true` and `false` (default). The

Reviewer 3: ```suggestion
  A boolean value that indicates whether a field is a ratio. Defaults to `false`. The
```

>   value should not be `true` when a `unit` is also specified. This information
>   may only be specified on numeric fields.

Reviewer 3: ```suggestion
  Leave set to `false` if a `unit` is specified. Only valid for numeric fields.
```
